code-challenges.js

```diff
@@ -2,35 +2,155 @@
+}
+
+console.log(charConversion(secretCodeWord1));
+console.log(charConversion(secretCodeWord2));
+
+
+/**
+  * VERSION 2.0
+  */
+
+const charConversionV2 = str => {
+  // store each unique key with its corresponding value
+  const chars = {"a": 4, "e": 3, "i": 1, "o": 0};
+  // replace() looks for a match using regex
+  // the anonymous function takes the matching letter and pass it to the chars object to access the value of that specific letter through bracket notation
+  // that value then replaces the matched letter thanks to the replace() method :)
+  return str.replace(/[aeio]/gi, letter => chars[letter])
+}
+
+console.log(charConversionV2(secretCodeWord1));
+console.log(charConversionV2(secretCodeWord2));
 
 
 
 
+// --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
+
 var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
 // Expected output: "Apple" "Banana" "Peach"
 
+// filter through the array
+// lowercase each word
+// use indexOf to check if 'a' is within the string
+const hasA = arr => {
+  return arr.filter(word => word.toLowerCase().indexOf('a') !== -1);
+}
 
+// includes return value is either true or false depending if the value is in the string
+const includesA = arr => {
+  return arr.filter(word => word.toUpperCase().includes('A'));
+}
 
+// when using the match() method, the argument passed in must be regex
+const matchesA = arr => {
+  return arr.filter(word => word.match(/[a]/i));
+}
 
-
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is one pair and one three of a kind.
+console.log(hasA(arrayOfWords));
+console.log(includesA(arrayOfWords));
+console.log(matchesA(arrayOfWords));
 
 
 
 
+// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is one pair and one three of a kind.
 
 var hand1 = [5, 5, 5, 3, 3]
 // Expected output: true
 var hand2 = [5, 5, 3, 3, 4]
 // Expected output: false
+
+/**
+  * VERSION 1.0
+  */
+
+const fullHouse = arr => {
+  // array to store unique elements
+  const temp = [];
+
+  // error check for length
+  if (arr.length > 5 ) {
+    return "Please return an array with only 5 elements."
+  }
+
+  arr.filter(num => {
+    // error check for numbers
+    if (typeof num !== "number") {
+      return "Only numbers allowed."
+    }
+    // if temp array doesn't include the current num
+    // push it to the temp array
+    // doesn't add duplicates since the next iteration check for presence in temp
+    else if (temp.includes(num) === false) {
+      temp.push(num);
+    }
+  })
+
+  // length of temp is 2 means there are 2 unique numbers within a 5 number array
+  // 3,3 & 5,5,5 >> 3 and 5
+  return temp.length === 2 ? true : false;
+}
+
+console.log(fullHouse(hand1));
+console.log(fullHouse(hand2));
```

Comment:
this solution returns true for a hand of `[4, 5, 5, 5, 5]` because i'm pretty sure it's just checking if 2 numbers exist in the hand. but great stab at a tough question!!